cloudinit/net/cmdline.py

```diff
@@ -72,7 +73,7 @@ def is_applicable(self) -> bool:
                 (ii) an open-iscsi interface file is present in the system
         """
         if self._files:
-            if 'ip=' in self._cmdline or 'ip6=' in self._cmdline:
+            if re.search(r'(^|\s)ip[6]*=', self._cmdline):
```

Comment:
I think `[6]*` is slightly incorrect, it would match, e.g., `ip666` (the addressing protocol of the beast); `6?` will match 0 or 1 "6"s. (Either way, the square brackets are superfluous: a character and a bracket expression containing only that character are equivalent.) This is borderline nit territory, but as regexes are hard enough to read at the best of times, I think it's worth being exact here.

I think we're also missing tests for the `^` choice of the subexpression here: nothing fails if I edit the regex to only have `\s` there instead of the subexpression. Can you add a test for that?

Reply:
I'm just flying past here, but you should really "parse" the kernel cmdline with shlex.split, because that's essentially what the initramfs is doing. This regex would match something like the admittedly extremely contrived `opt="bunch of stuff with spaces and ip=foo"`. Also the fewer regexes in the codebase the better!